pulp_rpm/tests/functional/api/test_publish.py

```diff
@@ -564,3 +564,49 @@ def do_test(self, url=None):
         self.addCleanup(self.distributions.delete, distribution.pulp_href)
 
         return distribution.to_dict()["base_url"]
+
+
+class PublishUnsupportedChecksumTestCase(PulpTestCase):
+    """Test unsupported publish."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = gen_rpm_client()
+        cls.repo_api = RepositoriesRpmApi(cls.client)
+        cls.remote_api = RemotesRpmApi(gen_rpm_client())
+        cls.publications = PublicationsRpmApi(cls.client)
+        cls.distributions = DistributionsRpmApi(cls.client)
+
+    def test_publish_with_unsupported_checksum_type(self):
+        """
+        Sync and and try publish an RPM repository.
+
+        This test require disallowed 'MD5' checksum type from ALLOWED_CONTENT_CHECKSUMS settings.
+        """
```

Comment:
Probably want to skip this test `if "md5" in utils.get_pulp_setting(cli_client, "ALLOWED_CONTENT_CHECKSUMS")`

Reply:
get_pulp_setting fails on get ALLOWED_CONTENT_CHECKSUMS as it tries json serialize on it (set is not supported by json). Looking into other option as I believe we can't import directly from django.conf.

Reply:
solved by https://github.com/pulp/pulp-smash/pull/1248